model/shop-ubytovani.php

```diff
@@ -80,18 +80,21 @@ function zpracuj() {
     if(!isset($_POST[$this->pnDny])) return false;
 
     // smazat veškeré stávající ubytování uživatele
-    dbQuery('
-      DELETE n.* FROM shop_nakupy n
-      JOIN shop_predmety p USING(id_predmetu)
-      WHERE n.id_uzivatele='.$this->u->id().' AND p.typ=2 AND n.rok='.ROK);
+    $deleteQuery = '
+    DELETE n.* FROM shop_nakupy n JOIN shop_predmety p USING(id_predmetu)
+    WHERE n.id_uzivatele='.$this->u->id().' AND p.typ=2 AND n.rok='.ROK;
+
+    dbQuery($deleteQuery);
 
     // vložit jeho zaklikané věci - note: není zabezpečeno
     $q = 'INSERT INTO shop_nakupy(id_uzivatele,id_predmetu,rok,cena_nakupni,datum) VALUES '."\n";
     foreach($_POST[$this->pnDny] as $predmet) {
       if(!$predmet) continue;
       $q .= '('.$this->u->id().','.(int)$predmet.','.ROK.',(SELECT cena_aktualni FROM shop_predmety WHERE id_predmetu='.(int)$predmet.'),NOW()),'."\n";
-      if($this->presKapacitu($predmet))
+      if($this->presKapacitu($predmet)) {
+        dbQuery($deleteQuery);
         Log::warning('Uživateli '.$this->u->jmenoNick().' bylo objednáno ubytování id '.$predmet.' přes kapacitu.');
+      }
```

Comment:
Jo, určitě tu chybí hláška pro uživatele. Něco jako "Vybrané ubytování je už bohužel zabrané. Vyber si prosím jiné."